deploy/kubernetes/nfs_services_start.sh

```diff
@@ -5,6 +5,19 @@ set -o errexit
 trap "{ exit 0 }" TERM
 
 service rpcbind start
-service nfs-common start
+
+# If statd is already running, for example becuase of an existing nfs mount, we will fail
+# to service nfs-common start. If we successfully query the statd service (rpc program
+# number 100024), that means it's running, and we don't need to start it. This command
+# is put in /etc/default/nfs-common as the NEED_STATD variable must be set there.
+
+if rpcinfo -T udp 127.0.0.1 100024; then
```

Comment:
I think there is an issue if "Program not resgistered" is the output of rpcinfo. thoughts?

1. $ sudo rpcinfo -T udp 127.0.0.1 status
rpcinfo: RPC: Program not registered

2. echo no to NEED_STATD
```
sudo cat /etc/default/nfs-common
# If you do not set values for the NEED_ options, they will be attempted
# autodetected; this should be sufficient for most people. Valid alternatives
# for the NEED_ options are "yes" and "no".

# Do you want to start the statd daemon? It is not needed for NFSv4.
NEED_STATD=no

# Options for rpc.statd.
#   Should rpc.statd listen on a specific port? This is especially useful
#   when you have a port-based firewall. To use a fixed port, set this
#   this variable to a statd argument like: "--port 4000 --outgoing-port 4001".
#   For more information, see rpc.statd(8) or http://wiki.debian.org/SecuringNFS
STATDOPTS=

# Do you want to start the idmapd daemon? It is only needed for NFSv4.
NEED_IDMAPD=

# Do you want to start the gssd daemon? It is required for Kerberos mounts.
NEED_GSSD=
```
3. start nfs-common and check status
```
$ sudo service nfs-common start
$ sudo service nfs-common status
● nfs-common.service - LSB: NFS support files common to client and server
   Loaded: loaded (/etc/init.d/nfs-common; generated)
   Active: active (running) since Tue 2021-06-22 07:37:45 UTC; 3s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 3818 ExecStart=/etc/init.d/nfs-common start (code=exited, status=0/SUCCESS)
    Tasks: 1 (limit: 4665)
   Memory: 1.0M
   CGroup: /system.slice/nfs-common.service
           └─3831 /usr/sbin/rpc.idmapd
```
we can see rpc.statd is missing

4. remove no from NEED_STATD
```
sudo cat /etc/default/nfs-common
# If you do not set values for the NEED_ options, they will be attempted
# autodetected; this should be sufficient for most people. Valid alternatives
# for the NEED_ options are "yes" and "no".

# Do you want to start the statd daemon? It is not needed for NFSv4.
NEED_STATD=

# Options for rpc.statd.
#   Should rpc.statd listen on a specific port? This is especially useful
#   when you have a port-based firewall. To use a fixed port, set this
#   this variable to a statd argument like: "--port 4000 --outgoing-port 4001".
#   For more information, see rpc.statd(8) or http://wiki.debian.org/SecuringNFS
STATDOPTS=

# Do you want to start the idmapd daemon? It is only needed for NFSv4.
NEED_IDMAPD=

# Do you want to start the gssd daemon? It is required for Kerberos mounts.
NEED_GSSD=
```

5.  restart nfs-common, now we have rpc.statd running:
```
$ sudo service nfs-common restart
$ $ sudo service nfs-common status
● nfs-common.service - LSB: NFS support files common to client and server
   Loaded: loaded (/etc/init.d/nfs-common; generated)
   Active: active (running) since Tue 2021-06-22 07:41:05 UTC; 3s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 4028 ExecStart=/etc/init.d/nfs-common start (code=exited, status=0/SUCCESS)
    Tasks: 2 (limit: 4665)
   Memory: 49.7M
   CGroup: /system.slice/nfs-common.service
           ├─4036 /sbin/rpc.statd
           └─4049 /usr/sbin/rpc.idmapd
```

6. 
```
$ sudo rpcinfo -T udp 127.0.0.1 status
program 100024 version 1 ready and waiting
```

Reply:
Sorry, I don't grok this. Let's sync offline.